src/core/io/src/4C_io_mesh.hpp

```diff
@@ -282,8 +282,15 @@ namespace Core::IO::MeshInput
         return raw_mesh_->external_ids ? (*raw_mesh_->external_ids)[index_] : invalid_external_id;
       }
 
+      /**
+       * Check if the point has data for the given field name.
+       */
+      [[nodiscard]] bool contains_data(const std::string& field_name) const
+      {
+        return raw_mesh_->point_data.contains(field_name);
+      }
```

Comment:
Having data is a property of the whole mesh and not just a point reference. We always assume that data is available for all points. I would move this function to the `Mesh` class.